### DISCUSSION TITLE
Auth library for distribution auth spec

### DISCUSSION THREAD

**Discussion by sagikazarmark** [distribution/distribution#3746] [Show and tell]
I started working on an auth library for the [auth spec](https://github.com/distribution/distribution/tree/main/docs/spec/auth) described in this repository: https://github.com/distribution-auth/auth

It's a bit rough around the edges, but the goal is to provide a toolkit that allows custom authorization server implementations.

I'll polish it in the coming weeks and will hopefully tag a first working version.

I was wondering if you'd be interested in making it an official/reference implementation once it reaches a level of maturity.

I know there is a token server implementation in the repository, but it's rather an example than a full-fledged implementation suitable for production usage.

Let me know what you think!

**Comment by milosgajdos** [comment]
FYI: There is a working group proposal in OCI https://github.com/opencontainers/tob/pull/119 so I think we should wait until that pans out so we don't duplicate this effort.

**Reply by sagikazarmark** [reply]
Makes sense, thanks for the link.
